src/test/java/org/owasp/validator/html/test/ESAPIInvalidPolicyTest.java

```diff
@@ -0,0 +1,76 @@
+package org.owasp.validator.html.test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.lang.reflect.Method;
+
+import org.junit.After;
+import org.junit.Test;
+import org.owasp.validator.html.Policy;
+
+public class ESAPIInvalidPolicyTest {
+	/**
+	 * Property specified by the Antisamy project which may be used to disable
+	 * schema validation on policy files.
+	 */
+	private static final String ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION = "owasp.validator.validateschema";
+
+	@After
+	public void resetSystemProp() throws Exception {
+		System.clearProperty(ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION);
+		reloadSchemaValidation();
+	}
+
+	@Test
+	public void testDirectConfigAsBais() throws Exception {
+		Policy.setSchemaValidation(false);
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(toByteArrayStream(stream));
+	}
+
+	@Test
+	public void testDirectConfig() throws Exception {
+		Policy.setSchemaValidation(false);
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(stream);
+	}
+
+	@Test
+	public void testSystemProp() throws Exception {
+
+		System.setProperty(ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION, "false");
+		reloadSchemaValidation();
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(stream);
+	}
+
+	@Test
+	public void testSystemPropAsBais() throws Exception {
+		System.setProperty(ANTISAMY_PROJECT_PROP_SCHEMA_VALIDATION, "false");
+		reloadSchemaValidation();
+		InputStream stream = ESAPIInvalidPolicyTest.class.getResourceAsStream("/esapi-antisamy-InvalidPolicy.xml");
+		Policy.getInstance(toByteArrayStream(stream));
```

Comment:
Please invoke resetSystemProp() at the end of this test case so as to not affect other test cases that are expecting the default validation ON behavior.

Reply:
Same case. The @after method takes care of that